client/errors.go

```diff
@@ -59,19 +58,6 @@ func (e objectNotFoundError) Error() string {
 	return fmt.Sprintf("Error: No such %s: %s", e.object, e.id)
 }
 
-func wrapResponseError(err error, resp serverResponse, object, id string) error {
-	switch {
-	case err == nil:
-		return nil
-	case resp.statusCode == http.StatusNotFound:
-		return objectNotFoundError{object: object, id: id}
-	case resp.statusCode == http.StatusNotImplemented:
-		return errdefs.NotImplemented(err)
```

Comment:
I 100% agree the 404 handling here is less than ideal (and can see how it would hide the error, since it's not part of the new object), but I worry about this `errdefs.NotImplemented` handling -- are there other users of this client that might be affected by this negatively and no longer accurately detect when the answer they get back is a "Not Implemented" answer?

Reply:
Ah, I'm satisfied! :heart: